src/lib/proxy-server.ts

```diff
@@ -134,42 +222,65 @@ export async function startProxyServer(): Promise< boolean > {
 				type: 'error',
 				message: __( 'Custom domain set up failed' ),
 				detail: __(
-					'Studio needs to use port 80, but it’s already in use by another app. Close any local development apps and restart Studio.'
+					'Studio needs to use port 80 and 443, but they are already in use by another app. Close any local development apps and restart Studio.'
 				),
 				buttons: [ __( 'OK' ) ],
 			} );
 			return false;
 		}
 
 		Sentry.captureException( error );
-		console.error( `Failed to start proxy server directly:`, error );
+		console.error( 'Failed to start proxy servers:', error );
 		return false;
 	}
 }
 
 /**
- * Stop the proxy server
+ * Stop the proxy servers
  */
 export function stopProxyServer(): Promise< void > {
 	return new Promise( ( resolve ) => {
-		if ( ! proxyServer ) {
-			isProxyRunning = false;
-			resolve();
-			return;
+		const promises: Promise< void >[] = [];
+
+		// Stop HTTP proxy if running
+		if ( httpProxyServer ) {
+			promises.push(
+				new Promise< void >( ( resolve ) => {
+					httpProxyServer!.close( () => {
+						httpProxyServer = null;
+						isHttpProxyRunning = false;
+						console.log( 'HTTP Proxy server stopped' );
+						resolve();
+					} );
+				} )
+			);
+		} else {
+			isHttpProxyRunning = false;
 		}
 
-		proxyServer.close( () => {
-			proxyServer = null;
-			isProxyRunning = false;
-			console.log( 'Proxy server stopped' );
-			resolve();
-		} );
+		// Stop HTTPS proxy if running
+		if ( httpsProxyServer ) {
+			promises.push(
+				new Promise< void >( ( resolve ) => {
+					httpsProxyServer!.close( () => {
+						httpsProxyServer = null;
+						isHttpsProxyRunning = false;
+						console.log( 'HTTPS Proxy server stopped' );
+						resolve();
+					} );
+				} )
+			);
+		} else {
+			isHttpsProxyRunning = false;
+		}
+
+		Promise.all( promises ).then( () => resolve() );
```

Comment:
```suggestion
		await Promise.all( promises );
```
Optional nit, but if we made the function async, we could do this instead.